web/src/settings_org.js

```diff
@@ -261,36 +261,68 @@ function set_msg_move_limit_setting(property_name) {
     enable_or_disable_related_message_move_time_limit_setting(property_name, disable_setting);
 }
 
-function message_delete_limit_setting_enabled(setting_value) {
+function message_delete_limit_setting_enabled() {
     // This function is used to check whether the time-limit setting
-    // should be enabled. The setting is disabled when delete_own_message_policy
-    // is set to 'admins only' as admins can delete messages irrespective of
-    // time limit.
-    if (setting_value === settings_config.common_message_policy_values.by_admins_only.code) {
+    // should be enabled. The setting is disabled when every user
+    // who is allowed to delete their own messages is also allowed
+    // to delete any message in the organization.
+    const realm_delete_own_message_policy = Number.parseInt(
+        $("#id_realm_delete_own_message_policy").val(),
+        10,
+    );
+    const realm_can_delete_any_message_group_id =
+        settings_components.get_dropdown_list_widget_setting_value(
+            $("#id_realm_can_delete_any_message_group"),
+        );
+    const realm_can_delete_any_message_group_name = user_groups.get_user_group_from_id(
+        realm_can_delete_any_message_group_id,
+    ).name;
+    const common_message_policy_values = settings_config.common_message_policy_values;
+
+    if (realm_delete_own_message_policy === common_message_policy_values.by_admins_only.code) {
+        return false;
+    }
+    if (realm_can_delete_any_message_group_name === "role:administrators") {
+        return true;
+    }
+    if (realm_delete_own_message_policy === common_message_policy_values.by_moderators_only.code) {
+        return false;
+    }
+    if (realm_can_delete_any_message_group_name === "role:moderators") {
+        return true;
+    }
+    if (realm_delete_own_message_policy === common_message_policy_values.by_full_members.code) {
+        return false;
+    }
+    if (realm_can_delete_any_message_group_name === "role:fullmembers") {
+        return true;
+    }
+    if (realm_delete_own_message_policy === common_message_policy_values.by_members.code) {
```

Comment:
This is somewhat hard to parse on a quick read, but works fine. I don't think we would be able to check this efficiently when we would allow any group to be set to setting, so should be fine for now.

Reply:
This is fine for now. I think we can migrate `realm_delete_own_message_policy` to the group-setting-value system as a near-term follow-up to this PR, and replace this messy code with an "does this group contain users not in this other group" check as part of that PR.